website/marietje/api/v1/views.py

```diff
@@ -0,0 +1,190 @@
+import spotipy
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from rest_framework.generics import ListAPIView, RetrieveAPIView
+from rest_framework.response import Response
+
+from marietje import services
+from marietje.api.v1.pagination import StandardResultsSetPagination
+from marietje.api.v1.serializers import PlayerSerializer, QueueItemSerializer
+from marietje.models import Player, SpotifyQueueItem
+
+
+class PlayerListAPIView(ListAPIView):
+    """
+    Player List API View.
+
+    Permissions required: None
+
+    Use this endpoint to get a list of all installed Spotify Players.
+    """
+
+    serializer_class = PlayerSerializer
+    queryset = Player.objects.all()
+
+
+class PlayerRetrieveAPIView(RetrieveAPIView):
+    """
+    Player Retrieve API View.
+
+    Permissions required: None
+
+    Use this endpoint to get the details of an installed Spotify Players.
+    """
+
+    serializer_class = PlayerSerializer
+    queryset = Player.objects.all()
+
+
+class PlayerQueueListAPIView(ListAPIView):
+    """
+    Player Queue List API View.
+
+    Permissions required: None
+
+    Use this endpoint to get a list of tracks that are in the queue of the Spotify Player. Tracks are sorted on the
+    time that they were added, the track that was added last will appear first in the list.
+    """
+
+    serializer_class = QueueItemSerializer
+    queryset = SpotifyQueueItem.objects.all()
+    pagination_class = StandardResultsSetPagination
+
+    def get_queryset(self):
+        """Get the queryset."""
+        return self.queryset.filter(player=self.kwargs.get("player"))
+
+
+@api_view(["GET"])
```

Comment:
If we could rewrite these to class-based views, you could add the parameters like the search query to the openapi schema so swagger would understand it (by setting `schema=` for the view)

Reply:
Done, only the Player schema is not displayed correctly but we might fix that later.